c/src/metta.rs

```diff
@@ -1412,6 +1412,24 @@ pub extern "C" fn grounded_number_to_longlong(n: *const atom_t, res: *mut c_long
     }
 }
 
+#[no_mangle]
+pub extern "C" fn grounded_bool_to_int(n: *const atom_t, res: *mut c_int) -> bool {
```

Comment:
`atom_t` and `atom_ref_t` are designed to be interchangeable, but across the C API, we always use `atom_ref_t` when taking a pointer to a constant atom.

Reply:
One of the reasons @Necr0x0Der  implemented it in this way is that I implemented it in this way earlier. And it is my fault, that I didn't add you to the review @luketpeterson. I agree with the comment and I think we need to make all three functions (two I introduced in #543 and one from this PR) look uniformly.

Reply:
Fixed in https://github.com/trueagi-io/hyperon-experimental/pull/544/commits/3a49aff93e765c634a1aeb5e9c79cce3ea5148d0